### PR TITLE
Revision hashes

### PR DESCRIPTION
**Pull request by phlax**
Add a model for storing revision hashes or numbers against either the site, a model or an instance of a model

also can have different keys for above

### CODE REVIEW

File: pootle/apps/pootle_data/directory_data.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ class DirectoryDataTool(RelatedStoresDataTool):
 
     @property
     def context_name(self):
-        return self.context.pootle_path
+        return self.context.pootle_path.strip("/").replace("/", ".")
```
Comment (ta2-1) [inline pootle/apps/pootle_data/directory_data.py]: can you recall why it is necessary? 
`/af/prj0/dir.dir/` can't be distinct from `/af/prj/dir/dir/`

File: pootle/apps/pootle_data/project_data.py [hunk 2]
```diff
@@ -20,22 +20,35 @@ def filter_data(self, qs):
...
 class ProjectSetDataTool(RelatedTPsDataTool):
     group_by = ("tp__project__code", )
+    cache_key_name = "projects"
 
     def get_root_child_path(self, child):
         return child[self.group_by[0]]
+
+    @property
+    def context_name(self):
+        return "ALL"
```
Comment (ta2-1) [inline pootle/apps/pootle_data/project_data.py]: why not `projects`?

File: pootle/apps/pootle_data/utils.py [hunk 3]
```diff
@@ -274,9 +274,11 @@ def aggregate_max_fields(self):
             for f
             in self.max_fields)
 
-    @property
+    @persistent_property
```
Comment (ta2-1) [inline pootle/apps/pootle_data/utils.py]: I see that revision updater in store_data save handler can update "checks" however I couldn't find where it is used.
IIUC persistent_property for checks should use "checks" key in rev_cache_key though I could misunderstand.

File: pootle/apps/pootle_data/directory_data.py (hunk 1 above)
Comment (phlax) [inline pootle/apps/pootle_data/directory_data.py]: fair point ill revert

File: pootle/apps/pootle_data/utils.py (hunk 3 above)
Comment (phlax) [inline pootle/apps/pootle_data/utils.py]: as they are updated together - removed the checks key for now

File: pootle/apps/pootle_data/project_data.py (hunk 2 above)
Comment (phlax) [inline pootle/apps/pootle_data/project_data.py]: the cache_key_name is projects

**Comment by unho** [comment]
lgtm

**Pull request merged**